### PR TITLE
Prevent flush on closed NamedStream

### LINKED ISSUE #3386: ReaderBase.__del__:  I/O operation on closed file
## Expected behavior ##

I expect MDAnalysis to terminate without (ignored) exceptions.

## Actual behavior ##

When I create an universe from a PDB file defined as `StringIO`, I get the following error:
```
Exception ignored in: <function ReaderBase.__del__ at 0x7ff8d00a8dd0>
Traceback (most recent call last):
  File "/Users/rmeli/Documents/git/software/mdanalysis/package/MDAnalysis/coordinates/base.py", line 2175, in __del__
  File "/Users/rmeli/Documents/git/software/mdanalysis/package/MDAnalysis/coordinates/PDB.py", line 458, in close
  File "/Users/rmeli/Documents/git/software/mdanalysis/package/MDAnalysis/lib/util.py", line 720, in close
  File "/Users/rmeli/Documents/git/software/mdanalysis/package/MDAnalysis/lib/util.py", line 826, in flush
ValueError: I/O operation on closed file.
```

## Code to reproduce the behavior ##

``` python
from io import StringIO

import MDAnalysis as mda

print(mda.__version__)

res_str = """\
ATOM    659  N   THR A 315      22.716  15.055  -1.000  1.00 16.08         A N
ATOM    660  CA  THR A 315      22.888  13.803  -0.302  1.00 15.13         A C
ATOM    661  C   THR A 315      22.006  12.700  -0.882  1.00 15.69         A C
ATOM    662  O   THR A 315      21.138  12.959  -1.727  1.00 16.25         A O
ATOM    663  CB  THR A 315      22.481  13.956   1.182  1.00 16.22         A C
ATOM    664  CG2 THR A 315      22.874  15.310   1.747  1.00 17.32         A C
ATOM    665  OG1 THR A 315      21.047  13.922   1.304  1.00 15.14         A O
"""

res = mda.Universe(StringIO(res_str), format='PDB')
```

## Current version of MDAnalysis ##

- Which version are you using? `2.0.0-dev0`
- Which version of Python? `3.7.10`
- Which operating system? `macOS 11.5.1`

### PR DESCRIPTION
**Pull request by IAlibay**
Fixes #3386

Changes made in this Pull Request:
 - Prevents trying to close (or seek) an already closed stream.

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

**Comment by IAlibay** [comment]
I think codecov is being weird again - not sure why it thinks there's two extra misses. We can add an issue to raise test coverage for lib later? (it's a simple enough GSoC-starter)

File: package/MDAnalysis/lib/util.py [hunk 1]
```diff
@@ -724,7 +724,15 @@ def close(self, force=False):
         .. Note:: This ``close()`` method is non-standard. ``del NamedStream``
                   always closes the underlying stream.
 
+
+        .. versionchanged:: 2.1.0
+           Calls to ``close()`` will no longer attempt to close or flush the
+           stream if :attr:`closed` is `True`.
```
Comment (tylerjereddy) [inline package/MDAnalysis/lib/util.py]: Won't it get pretty verbose if you start adding `versionchanged` directives for bug fixes as opposed to select new features or API changes?

Comment (IAlibay) [reply package/MDAnalysis/lib/util.py]: Yeah I can see that. I'm happy to not include that, I'm tempted to remove the entry from the changelog too if you're good with that?

**Review by richardjgowers — APPROVED** [review]
Lgtm

**Comment by jbarnoud** [comment]
Is there anything else than the buggy codecov that prevents this PR from going through?

**Comment by IAlibay** [comment]
Mostly forgot about it 😅

I did wonder if anyone else had any strong opinions about what is essentially hiding some of the `__del__` issues we have.

**Review by jbarnoud — APPROVED** [review]

**Comment by jbarnoud** [comment]
Thank you @IAlibay :smile:

**Pull request merged**